### DISCUSSION TITLE
How to retrieve the first message of a forum post

### DISCUSSION THREAD

**Discussion by AlexCai2019** [discord-jda/JDA#2424] [Questions and Help] [answered]
I tried using `forumPost.retrieveParentMessage()`, but because the parent channel of a forum post is not a MessageChannel but a ForumChannel, hence it throws an `UnsupportedOperationException`.
Here is part of the code that throws an exception:
```java
private static final Emoji reminder_ribbon = Emoji.fromUnicode("🎗️");
private static final String reminder_ribbonFormat = reminder_ribbon.getFormatted();

public static void archiveForumPost(ThreadChannel forumPost)
{
	forumPost.retrieveParentMessage().queue(message ->
	{
		if (message.getReactions().stream().anyMatch(messageReaction -> messageReaction.getEmoji().getFormatted().equals(reminder_ribbonFormat)))
			message.removeReaction(reminder_ribbon, forumPost.getJDA().getSelfUser()).queue();
	});
}
```

**Comment by MinnDevelopment** [comment, marked as answer]
You can use [getIterableHistory](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/middleman/MessageChannel.html#getIterableHistory()):

```java
channel.getIterableHistory()
  .reverse() // iterate from the beginning of the channel history
  .limit(1) // only take first message
  .queue(messages -> {
    Message message = messages.get(0); // get first from list (might throw if channel is empty)
  });
```

**Comment by AlexCai2019** [comment]
Update: after [v5.0.0-beta.10](https://github.com/discord-jda/JDA/releases/tag/v5.0.0-beta.10), you can use [retrieveStartMessage](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/channel/concrete/ThreadChannel.html#retrieveStartMessage()):
```java
channel.retrieveStartMessage().queue(message ->
{
  ///TODO: put your code here.
});
```

**Reply by MinnDevelopment** [reply]
Note that this is not necessarily the *first* message. The start message can be deleted, in which case this request fails, while a history request will just return the first message in history.
